### DISCUSSION TITLE
下载微信账单问题

### DISCUSSION THREAD

**Discussion by yqsphp** [yansongda/pay#881] [General]
通过加载插件请求对账单
`/申请交易账单
        $plugin = Pay::wechat()->mergeCommonPlugins([GetTradeBillPlugin::class]);
        //$plugin = Pay::wechat()->mergeCommonPlugins([DownloadBillPlugin::class]);
        
        $params = [
            'bill_date' => $date,
            'bill_type' => 'ALL',
        ];
        $result = Pay::wechat()->pay($plugin, $params);`
返回下载地址后，在调用插件
`$plugin = Pay::wechat()->mergeCommonPlugins([DownloadBillPlugin::class]);
        
        $params = [
            'download_url' => $result->download_url,
            //'sign' => $result->hash_value
        ];
        $bill = Pay::wechat()->pay($plugin, $params);

[5003] InvalidResponseException in Functions.php line 204
        $random = $message->getHeaderLine('Wechatpay-Nonce');
        $sign = $message->getHeaderLine('Wechatpay-Signature');
        $body = (string) $message->getBody();
        dump($sign);

        $content = $timestamp."\n".$random."\n".$body."\n";
        $public = get_wechat_config($params)['wechat_public_cert_path'][$wechatSerial] ?? null;

        if (empty($sign)) {
            throw new InvalidResponseException(Exception::INVALID_RESPONSE_SIGN, '', ['headers' => $message->getHeaders(), 'body' => $body]);
        }
`
异常，签名空，不知道是不是这么调用

**Comment by yansongda** [comment]
别验证签名。

Pay::wechat()->pay([
PreparePlugin::class,
DownloadBillPlugin::class,
RadarSignPlugin::class,
ParserPlugin::class,
], $params);
